Scripts/Runtime/Entities/Tasks/Util/ResolveChannelUtil.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Diagnostics;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    internal static class ResolveChannelUtil
+    {
+        //*************************************************************************************************************
+        // SAFETY
+        //*************************************************************************************************************
+
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+        public static void Debug_EnsureEnumValidity<TResolveChannel>(TResolveChannel resolveChannel)
+            where TResolveChannel : Enum
+        {
+            Debug_EnsureEnumIsSizedProperly(typeof(TResolveChannel));
+        }
+        
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+        public static void Debug_EnsureEnumValidity(object resolveChannel)
+        {
+            Type type = resolveChannel.GetType();
+            if (!type.IsEnum)
+            {
+                throw new InvalidOperationException($"Resolve Channel Type is {type} but needs to be a {typeof(Enum)}");
+            }
+
+            Debug_EnsureEnumIsSizedProperly(type);
+        }
+        
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+        private static void Debug_EnsureEnumIsSizedProperly(Type type)
+        {
+            int sizeOfType = UnsafeUtility.SizeOf(type);
+            int sizeOfByte = UnsafeUtility.SizeOf<byte>();
+            
```

Comment:
Do you also need to ensure that it's Assignable to `byte`?

Reply:
It's not assignable, that's only for inheritance. But I can check via:

`if (Enum.GetUnderlyingType(type) != BYTE_TYPE)`

so now I am.